contracts/EscrowAndStaking/CrossChainCollector.sol

```diff
@@ -0,0 +1,441 @@
+
+import {IRouterClient} from "@chainlink/contracts-ccip/src/v0.8/ccip/interfaces/IRouterClient.sol";
+import {Client} from "@chainlink/contracts-ccip/src/v0.8/ccip/libraries/Client.sol";
+// import {LinkTokenInterface} from "@chainlink/contracts/src/v0.8/shared/interfaces/LinkTokenInterface.sol";
+// import {CCIPReceiver} from "@chainlink/contracts-ccip/src/v0.8/ccip/applications/CCIPReceiver.sol";
+
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+
+// internal
+import "../utils/proxy/solidity-0.8.0/ProxyReentrancyGuard.sol";
+import "../utils/proxy/solidity-0.8.0/ProxyOwned.sol";
+import "../utils/proxy/solidity-0.8.0/ProxyPausable.sol";
+
+import "../interfaces/IStakingThales.sol";
+
+import "./CCIPReceiverProxy.sol";
+
+/// @title - Cross Chain Collector contract for Thales staking rewards
+contract CrossChainCollector is Initializable, ProxyOwned, ProxyPausable, ProxyReentrancyGuard, CCIPReceiverProxy {
+    // Custom errors to provide more descriptive revert messages.
+    error NotEnoughBalance(uint256 currentBalance, uint256 calculatedFees); // Used to make sure contract has enough balance.
+    uint private constant MAX_GAS = 10 * 1e6;
+    uint private constant ONE_MILLION_GAS = 1e6;
+    uint private constant ONE = 1e18;
+    IRouterClient private s_router;
+    address public stakingThales;
+
+    address public masterCollector;
+    uint64 public masterCollectorChain;
+
+    mapping(uint => uint64) public chainSelector;
+    mapping(uint64 => uint) public chainSelectorIndex;
+    mapping(uint => address) public collectorAddress;
+
+    mapping(uint => uint) public lastPeriodForChain;
+    mapping(uint => mapping(uint => uint)) public chainStakedAmountInPeriod;
+    mapping(uint => mapping(uint => uint)) public chainEscrowedAmountInPeriod;
+    mapping(uint => mapping(uint => uint)) public chainBaseRewardsInPeriod;
+    mapping(uint => mapping(uint => uint)) public chainExtraRewardsInPeriod;
+    mapping(uint => mapping(uint => uint)) public chainBonusPointsInPeriod;
+    mapping(uint => mapping(uint => uint)) public chainRevenueInPeriod;
+    mapping(uint => mapping(uint => uint)) public chainRevenueShareInPeriod;
+
+    mapping(uint => uint) public calculatedStakedAmountForPeriod;
+    mapping(uint => uint) public calculatedEscrowedAmountForPeriod;
+    mapping(uint => uint) public calculatedBonusPointsForPeriod;
+    mapping(uint => uint) public calculatedRevenueForPeriod;
+
+    mapping(bytes32 => bool) public messageIdAlreadyReceived;
+    mapping(uint => uint64) public messagesReceivedFromChainSelector;
+    mapping(uint => bytes) public messagesReceived;
+    uint public numOfMessagesReceived;
+
+    uint public numOfActiveCollectors;
+    uint public period;
+    uint public baseRewardsPerPeriod;
+    uint public extraRewardsPerPeriod;
+    uint public numOfCollectedResultsForThisPeriod;
+    uint public lastPeriodBeforeTesting;
+
+    bool public readyToBroadcast;
+    bool public readOnlyMode;
```

Comment:
Please delete everything related to readOnlyMode and testing

Reply:
code in functions deleted, variable set to private